src/redisearch_rs/rqe_iterators/tests/wildcard.rs

```diff
@@ -0,0 +1,241 @@
+
+    // Read some documents
+    for _i in 1..=3 {
+        let result = it.read();
+        assert!(result.is_ok());
+        assert!(result.unwrap().is_some());
+    }
+
+    assert_eq!(it.last_doc_id(), 3);
+
+    // Rewind
+    it.rewind();
+
+    // Check state after rewind
+    assert_eq!(it.last_doc_id(), 0);
+    assert!(!it.at_eof());
+
+    // Should be able to read from beginning again
+    let result = it.read();
+    assert!(result.is_ok());
+
+    let result = result.unwrap();
+    assert!(result.is_some());
+
+    let doc = result.unwrap();
+    assert_eq!(doc.doc_id, 1);
+    assert_eq!(it.last_doc_id(), 1);
+}
+
+#[test]
+fn read_after_skip() {
+    let mut it = Wildcard::new(10);
+
+    // Skip to middle
+    let result = it.skip_to(5);
+    assert_skip_to_found(result, 5);
+    assert_eq!(it.last_doc_id(), 5);
+
+    // Continue reading sequentially from 6 to 10
+    for expected_id in 6..=10 {
+        let result = it.read();
+        assert!(result.is_ok());
+
+        let result = result.unwrap();
+        assert!(result.is_some());
+
+        let doc = result.unwrap();
+        assert_eq!(doc.doc_id, expected_id);
+        assert_eq!(it.last_doc_id(), expected_id);
+    }
+
+    // After reading all remaining docs, should return EOF
+    let result = it.read();
+    assert!(result.is_ok());
+    assert!(result.unwrap().is_none());
+    assert!(it.at_eof());
+}
+
+#[test]
+fn skip_to_after_eof() {
+    let mut it = Wildcard::new(10);
+
+    // First, move to EOF by skipping beyond range
+    let result = it.skip_to(11);
+    assert!(result.is_ok());
+    assert!(it.at_eof());
+
+    // Try to skip to a valid target while at EOF
+    let result = it.skip_to(5);
+    assert!(result.is_ok());
+
+    let outcome = result.unwrap();
+    assert!(outcome.is_none());
+    assert!(it.at_eof());
+}
+
+#[test]
+fn zero_documents() {
+    let mut it = Wildcard::new(0);
+
+    // Should immediately be at EOF
+    assert!(it.at_eof(), "iterator with top_id=0 should be at EOF");
+    assert_eq!(it.last_doc_id(), 0, "last_doc_id should be 0");
+    assert_eq!(it.num_estimated(), 0, "num_estimated should be 0");
+
+    // Read should return None
+    let result = it.read();
+    assert!(result.is_ok(), "read() should succeed");
+    assert!(
+        result.unwrap().is_none(),
+        "read() should return None for empty iterator"
+    );
+
+    // Skip should return None
+    let result = it.skip_to(1);
+    assert!(result.is_ok(), "skip_to(1) should succeed");
+    assert!(
+        result.unwrap().is_none(),
+        "skip_to(1) should return None for empty iterator"
+    );
+}
+
+#[test]
+#[cfg(debug_assertions)]
+fn skip_to_backwards() {
+    let mut it = Wildcard::new(100);
+
+    let _ = it.skip_to(75);
+
+    // Try to skip backwards to position 25, should panic
+    std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
+        let _ = it.skip_to(25);
+    }))
+    .expect_err("Expected panic when skipping backwards");
+}
```

Comment:
```suggestion
#[test]
#[should_panic(expected = "The expected panic message")]
fn skip_to_backwards() {
    let mut it = Wildcard::new(100);

    let _ = it.skip_to(75);

    // Try to skip backwards to position 25, should panic
    let _ = it.skip_to(25);
}
```
The idiomatic way to test something panics with a given message